aws/resource_aws_eip_association_test.go

```diff
@@ -437,7 +421,35 @@ resource "aws_eip_association" "test" {
 
 func testAccAWSEIPAssociationConfig_spotInstance(rInt int) string {
 	return composeConfig(
-		testAccAWSSpotInstanceRequestConfig(rInt), `
+		testAccLatestAmazonLinuxHvmEbsAmiConfig(),
+		testAccAvailableEc2InstanceTypeForRegion("t3.micro", "t2.micro"),
+		testAccAvailableAZsNoOptInConfig(),
+		fmt.Sprintf(`
+resource "aws_default_subnet" "default" {
+  availability_zone = data.aws_availability_zones.available.names[0]
+
+  lifecycle {
+    ignore_changes = [
+      # testing environments often change the Name tag
+      tags["Name"],
+    ]
+  }
+}
```

Comment:
While we are in the neighborhood and adjusting the subnet handling, it'd be great to fix these to not use the default subnets here and below 😄 

```terraform
resource "aws_vpc" "test" {
  cidr_block = "10.0.0.0/16"
}

resource "aws_subnet" "test" {
  availability_zone = data.aws_availability_zones.available.names[0]
  cidr_block        = cidrsubnet(aws_vpc.test.cidr_block, 8, 0) 
  vpc_id            = aws_vpc.test.id
}

# ...

resource "aws_spot_instance_request" "test" {
  subnet_id            = aws_subnet.test.id
}
```